src/sql/parts/notebook/models/notebookContexts.ts

```diff
@@ -123,15 +124,14 @@ export class NotebookContexts {
 	/**
 	 *
 	 * @param specs kernel specs (comes from session manager)
-	 * @param connectionInfo connection profile
-	 * @param savedKernelInfo kernel info loaded from
+	 * @param displayName kernel info loaded from
 	 */
-	public static getDefaultKernel(specs: nb.IAllKernels, connectionInfo: IConnectionProfile, savedKernelInfo: nb.IKernelInfo): nb.IKernelSpec {
+	public static getDefaultKernel(specs: nb.IAllKernels, displayName: string): nb.IKernelSpec {
 		let defaultKernel: nb.IKernelSpec;
 		if (specs) {
 			// find the saved kernel (if it exists)
-			if (savedKernelInfo) {
-				defaultKernel = specs.kernels.find((kernel) => kernel.name === savedKernelInfo.name);
+			if (displayName) {
+				defaultKernel = specs.kernels.find((kernel) => kernel.display_name === displayName);
```

Comment:
I'm a bit worried this will fail for Python3. Please check it on Windows & Mac to see if it's different anywhere and causes bugs.